uniffi_bindgen/src/interface/mod.rs

```diff
@@ -241,6 +241,40 @@ impl<'ci> ComponentInterface {
         }
     }
 
+    /// Identify if we are using an unsigned type in kotlin
+    ///
+    /// In Kotlin, unsigned types are expiremental and will cause a warning if used without annotations
+    /// This should be removed when kotlin version
+    ///  > v1.5 https://kotlinlang.org/docs/whatsnew15.html#stable-unsigned-integer-types
+    pub fn contains_unsigned_type(&self, type_: &Type) -> bool {
+        match type_ {
+            Type::UInt8 => true,
+            Type::UInt16 => true,
+            Type::UInt32 => true,
+            Type::UInt64 => true,
```

Comment:
> Adding a quick review just because I got a bit of time and was curious, it looks good to me (not that I would be able to catch any big red flags lol)
> 
> I am curious about the whole Kotlin 1.5 thou, like at what point will we be able to get rid of those annotations? I'm not quite sure about how many Android versions backward our consumers support, I wonder if we should put a disclaimer somewhere (or at least for our own sanity) what the lowest version of Kotlin we support

Good question, I think it's tricky since the major downstream consumers for kotlin is a-c & Fenix and we probably won't be able to update to kotlin 1.5 until Fenix itself does to prevent any incompatibility.

I do think that's a good idea to keep a list of version compatibilities! It'll def help give context why certain things are done certain ways  -- I was also thinking about making a task to revisit this once Fenix updates their kotlin version but we'll see if this PR is even a good idea currently